### PR TITLE
WoW-style Autorun

### PR DESCRIPTION
**Pull request by duane-r**
This allows the player to toggle continuous forward with a key (F, by
default), so we don't have to hold down the forward key endlessly.

### CODE REVIEW

**Comment by PilzAdam** [comment]
Is this feature really that useful? You have to pay attention and control the walking anyway, or you'll get stuck on nodes easily.

**Comment by nerzhul** [comment]
:+1:

**Comment by duane-r** [comment]
Well, naturally I think it's terrific. The makers of World of Warcraft thought it was worthwhile. Using forward requires constant pressure with one digit, which is harder for some. In WoW, of course, you could map autorun and jump to your mouse button as well, for mouse-only control. Anyway, it seems silly to have the feature available in code, but not mapped to a key in case someone wants it.

**Comment by everamzah** [comment]
The maker of Quake 3 thought strafe jumping was worthwhile, I'm just saying.

**Comment by paramat** [comment]
+1 very useful for dev work on mapgen when flying, or long journeys on foot or in a vehicle.

**Comment by sfan5** [comment]
:+1:

I doulbt this is useful for *that* many people, but as long as it's disabled by default (to avoid confusion) it's fine.

**Comment by sfan5** [comment]
One thing, though:
Your code contains stuff for a <code>disable_mouse_wheel</code> setting, either split the commit into two commits or remove that code and make a new PR for that feature.

File: src/game.cpp [hunk 1]
```diff
@@ -2616,6 +2619,10 @@ void Game::processKeyboardInput(VolatileRunFlags *flags,
 
 	if (input->wasKeyDown(keycache.key[KeyCache::KEYMAP_ID_DROP])) {
 		dropSelectedItem();
+	// Add WoW-style autorun by toggling continuous forward.
+	} else if (input->wasKeyDown(keycache.key[KeyCache::KEYMAP_ID_AUTORUN])) {
+		bool autorun_setting = g_settings->getBool("continuous_forward");
```
Comment (kwolekr) [inline src/game.cpp]: There needs to be a default setting for continuous_forward or else this will crash minetest.

Comment (duane-r) [inline src/game.cpp]: There's one in defaultsettings.cpp already.

https://github.com/minetest/minetest/blob/master/src/defaultsettings.cpp#L121

**Comment by lightonflux** [comment]
Would be nice if the code was merged. This would help very much on servers with long walking distances.

**Comment by PilzAdam** [comment]
:+1: 

Maybe add this little status text when toggling that is also displayed when changing fly and fast mode?

**Comment by 0-afflatus** [comment]
:+1:

**Comment by sfan5** [comment]
b6dfae02215d7bd61e06ff1999ee3ecb0ec80b9e

**Pull request closed without merging**